server/asset/eth/coiner.go

```diff
@@ -55,30 +55,72 @@ func (be *AssetBackend) newSwapCoin(coinID []byte, contractData []byte) (*swapCo
 		return nil, err
 	}
 
-	inits, err := dexeth.ParseInitiateData(bc.txData, ethContractVersion)
-	if err != nil {
-		return nil, fmt.Errorf("unable to parse initiate call data: %v", err)
-	}
+	var sum uint64
+	var vector *dexeth.SwapVector
+	switch bc.contractVer {
+	case 0:
+		var secretHash [32]byte
+		copy(secretHash[:], bc.locator)
 
-	init, ok := inits[bc.secretHash]
-	if !ok {
-		return nil, fmt.Errorf("tx %v does not contain initiation with secret hash %x", bc.txHash, bc.secretHash)
-	}
+		inits, err := dexeth.ParseInitiateDataV0(bc.txData)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse v0 initiate call data: %v", err)
+		}
 
-	if be.assetID == be.baseChainID {
-		sum := new(big.Int)
+		init, ok := inits[secretHash]
+		if !ok {
+			return nil, fmt.Errorf("tx %v does not contain v0 initiation with secret hash %x", bc.txHash, secretHash[:])
+		}
 		for _, in := range inits {
-			sum.Add(sum, in.Value)
+			sum = +be.atomize(in.Value)
 		}
-		if bc.value.Cmp(sum) < 0 {
-			return nil, fmt.Errorf("tx %s value < sum of inits. %d < %d", bc.txHash, bc.value, sum)
+
+		vector = &dexeth.SwapVector{
+			// From: ,
+			To:         init.Participant,
+			Value:      init.Value,
+			SecretHash: secretHash,
+			LockTime:   uint64(init.LockTime.Unix()),
+		}
+
+		// if value < bc.vector.Value {
```

Comment:
The value check is moved and the others are completely pointless since we just pulled (v0) or validated (v1) the data on-chain. This is also not the purview of the backend, which reports chain data to the swapper for validation.